### PR TITLE
feat: add CD pipeline

### LINKED ISSUE #168: Load balancer Github Actions deployment pipeline
Create a github workflow that deploys the app when merged to `master`.

Currently, the deployment is done manually in an ubuntu server. We need to pull the `master` changes, `yarn buld` and kill the app to `yarn start` again. This stops the app from working a few seconds. 

We need a load balancing solution that will never turn off the app and that it automates the deployment once a merge to `master` is made.

### PR DESCRIPTION
**Pull request by Udit-takkar**
Fixes: https://github.com/aufacicenta/pulsemarkets/issues/168

### CODE REVIEW

**Comment by Udit-takkar** [comment]
@netpoe what platform do you use for deploying to production?

**Comment by Udit-takkar** [comment]
and can you tell me what steps do you follow to deploy this web app currently?

**Comment by netpoe** [comment]
> and can you tell me what steps do you follow to deploy this web app currently?

I do it manually, since Vercel only provides temporary server functions, parts of the game are automated: https://github.com/aufacicenta/pulsemarkets/blob/master/app/src/pages/api/prompt-wars/create.ts#L72-L94

I needed to keep the server alive so that scheduled API calls went through. 

I use a simple Ubuntu server, via SSH and keep a tmux session alive with an nginx server.

File: app/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+RUN_DIR=$(pwd -LP)
+#the presence of a .env file overrides the production file, which overrides staging, which overrides local
+source ../../.env.local
+source ../../.env.staging
+source ../../.env.production  
+source  ../../.env
+
+
```
Comment (Udit-takkar) [inline app/deploy.sh:9]: @netpoe what do you think about this script? I think this should work fine

I searched a lot and found this https://github.com/rekliner/next-pm2-deploy/tree/main

Comment (Udit-takkar) [reply app/deploy.sh:9]: one step remaining is configuring Configure Nginx as a reverse proxy

Comment (netpoe) [reply app/deploy.sh:9]: Also, I am not against Vercel. I still pay for deploying the preview app versions. Perhaps we should think differently regarding the automated API calls that run the game stages?

Let's study the `create.ts`   API endpoint and its subsequent timed API calls.

**Comment by netpoe** [comment]
@Udit-takkar 
photo_2023-11-05_22-10-57

Hey Udit, I sent another 300 USDT/NEAR to your wallet. While this CI/CD implementation is not a priority, we appreciate your continuous effort on building with us. 

Currently, as you may have noticed, we are focused on moving the game into an EVM compatible infrastructure. This will be our next step in the upcoming months. 

Perhaps, given your knowledge of Typescript and NextJS, you'd want to work on refactoring or updating the current NextJS app to work with the new Solidity ABIs. I'll keep you updated on the progress. We expect to have this Solidity implementation by the end of November. 

In the meantime, there are a couple of enhancements we want to implement to the current game UX. I'll creaste the bounties and let you know.

**Comment by Udit-takkar** [comment]
> Perhaps, given your knowledge of Typescript and NextJS, you'd want to work on refactoring or updating the current NextJS app to work with the new Solidity ABIs. I'll keep you updated on the progress. We expect to have this Solidity implementation by the end of November.

Yeah sure

**Pull request closed without merging**